app/controllers/api/v0/admins_controller.rb

```diff
@@ -1,14 +1,19 @@
 module Api
   module V0
     class AdminsController < BaseController
+      before_action :validate_admin
+
       def add_portfolio
         portfolio = Portfolio.create!(portfolio_params)
         render json: portfolio
+      rescue ActiveRecord::RecordInvalid => e
+        render :json => { :errors => e.message }, :status => :unprocessable_entity
       end
 
       def add_portfolio_item_to_portfolio
-        portfolio = Portfolio.find_by(id: params[:portfolio_id])
-        render json: portfolio.add_portfolio_item(params[:portfolio_item_id])
+        portfolio = Portfolio.find(params[:portfolio_id])
+        portfolio_item = PortfolioItem.find(params[:portfolio_item_id])
```

Comment:
Prefer Strong Params

Reply:
@bdunne How would strong parameters validate that the object with those ids exist. I was under the impression that strong parameters just checks for the presence of parameter values but doesn't check if they exist in the database. With the find calls I am testing the 404 being returned if the record is not found

Reply:
> How would strong parameters validate that the object with those ids exist

It won't, but it can require the parameter.

Reply:
@bdunne changed to use require